apollo-router-core/src/spec/query.rs

```diff
@@ -432,120 +615,181 @@ mod tests {
+                    "id": "1",
+                }
             }},
         );
     }
 
     #[test]
     fn reformat_response_data_fragment_spread() {
+        let schema = "type Query {
+          thing: Thing    
+        }
+
+        type Foo {
+            foo: String
+        }
+        type Bar {
+            bar: String
+        }
+        type Baz {
+            baz: String
+        }
+        union Thing = Foo
+        extend union Thing = Bar | Baz
+
+        fragment baz on Baz {baz}";
+        let query = "query { thing {...foo ...bar ...baz} } fragment foo on Foo {foo} fragment bar on Bar {bar}";
+
         assert_format_response!(
-            "fragment baz on Baz {baz}",
-            "{...foo ...bar ...baz} fragment foo on Foo {foo} fragment bar on Bar {bar}",
+            schema,
+            query,
             json! {
-            {"__typename": "Foo", "foo": "1", "bar": "2", "baz": "3"}
+                {"thing": {"__typename": "Foo", "foo": "1", "bar": "2", "baz": "3"}}
             },
             None,
             json! {
-                {"foo": "1"}
+                {"thing": {"foo": "1"}}
             },
         );
         assert_format_response!(
-            "fragment baz on Baz {baz}",
-            "{...foo ...bar ...baz} fragment foo on Foo {foo} fragment bar on Bar {bar}",
+            schema,
+            query,
             json! {
-            {"__typename": "Bar", "foo": "1", "bar": "2", "baz": "3"}
+                {"thing": {"__typename": "Bar", "foo": "1", "bar": "2", "baz": "3"}}
             },
             None,
             json! {
-                {"bar": "2"}
+                {"thing": {"bar": "2"} }
             },
         );
         assert_format_response!(
-            "fragment baz on Baz {baz}",
-            "{...foo ...bar ...baz} fragment foo on Foo {foo} fragment bar on Bar {bar}",
+            schema,
+            query,
             json! {
-            {"__typename": "Baz", "foo": "1", "bar": "2", "baz": "3"}
+                {"thing": {"__typename": "Baz", "foo": "1", "bar": "2", "baz": "3"}}
             },
             None,
             json! {
-                {"baz": "3"}
+                {"thing": {"baz": "3"} }
             },
         );
     }
 
     #[test]
     fn reformat_response_data_best_effort() {
         assert_format_response!(
-            "",
-            "{foo stuff{bar baz} ...fragment array{bar baz} other{bar}}",
+            "type Query {
+                get: Thing
+            }
+            type Thing {
+                foo: String
+                stuff: Baz
+                array: [Element]
+                other: Bar
+            }
+
+            type Baz {
+                bar: String
+                baz: String
+            }
+
+            type Bar {
+                bar: String
+            }
+
+            union Element = Baz | Bar | String
+            ",
+            "{get {foo stuff{bar baz} ...fragment array{bar baz} other{bar}}}",
```

Comment:
I think changing the query to: `"{get {foo stuff{bar baz} ...fragment array{bar ... on Baz {baz}} other{bar}}}"` should produce a different result but doesn't? Could be wrong, but goal was to have one element of the array be populated.